crates/rpc/rpc/src/trace.rs

```diff
@@ -39,6 +39,33 @@ pub struct TraceApi<Provider, Eth> {
     inner: Arc<TraceApiInner<Provider, Eth>>,
 }
 
+trait TransactionCallback<R>
+where
+    Self: for<'a> Fn(
+            TransactionInfo,
+            TracingInspector,
+            ExecutionResult,
+            &'a State,
+            &'a CacheDB<StateProviderDatabase<StateProviderBox<'a>>>,
+        ) -> EthResult<R>
+        + Send
+        + 'static,
+{
+}
+
+impl<F, R> TransactionCallback<R> for F where
+    F: for<'a> Fn(
+            TransactionInfo,
+            TracingInspector,
+            ExecutionResult,
+            &'a State,
+            &'a CacheDB<StateProviderDatabase<StateProviderBox<'a>>>,
+        ) -> EthResult<R>
+        + Send
+        + 'static
+{
+}
```

Comment:
I don't really like this because this is just a hack to make signatures nicer and not really needed